src/object/cli_obj.c

```diff
@@ -5140,6 +5141,11 @@ obj_csum_update(struct dc_object *obj, daos_obj_update_t *args, struct obj_auxi_
 	if (!obj_csum_dedup_candidate(&obj->cob_co->dc_props, args->iods, args->nr))
 		return 0;
 
+	if (obj_auxi->csum_retry) {
+		/* Release old checksum result and prepare for new calculation */
+		daos_csummer_free_ic(obj->cob_co->dc_csummer, &obj_auxi->rw_args.iod_csums);
```

Comment:
I think we probably want to do this after a couple of retries

Reply:
It's really easy to add but I wonder if that is indeed necessary, because cksum error is a rare event by itself.

How about revising it to:
```
if (obj_auxi->csum_retry && obj_auxi->csum_retry_cnt > 2) { ... }
```
would that work for you?

Reply:
yeah